src/features/encoders/avif/client/index.tsx

```diff
@@ -55,34 +64,30 @@ export class Options extends Component<Props, State> {
     const { options } = props;
 
     const lossless =
-      options.cqLevel === 0 &&
-      options.cqAlphaLevel <= 0 &&
+      options.quality === MAX_QUALITY &&
+      (options.qualityAlpha == -1 || options.qualityAlpha == MAX_QUALITY) &&
       options.subsample == 3;
 
-    const separateAlpha = options.cqAlphaLevel !== -1;
+    const separateAlpha = options.qualityAlpha !== -1;
 
-    const cqLevel = lossless ? defaultOptions.cqLevel : options.cqLevel;
+    const quality = lossless ? defaultOptions.quality : options.quality;
```

Comment:
I may have asked you about this line before, but I still don't understand this line (specifically, why do we use `defaultOptions.quality` if `lossless` is true). It would be good to add a comment.

Also, `defaultOptions.quality` seems to be 50 (see src/features/encoders/avif/shared/meta.ts). The quality for lossless should be 100, not 50.

Reply:
By default, we don't use the lossless mode so we keep the quality at a _reasonable_ 50. Agreed, using `defaultOptions.quality` seems wrong here.

<hr>

@jakearchibald correct me if I'm wrong: There are are two situations where we want to generate the encoding state:

1. When the component is mounted for the first time - we use `getDerivedStateFromProps` to generate the state as a function of the props (which will be `defaultOptions`)
2. When any input changes - we use the `_inputChange` function to set the new encoding state.

So yeah I think we can completely remove `defaultOptions` from this file because it'll be equivalent to using `props.options`. Also, 

```typescript
const quality = lossless ? defaultOptions.quality : options.quality;
```

can be removed? Because `lossless` will be true iff `options.quality` is 100 anyway.

Reply:
Remember that this is UI state, not encoder state.

What if:

1. The user changes the quality slider to 30.
2. The user checks the lossless checkbox (removing the quality slider).
3. The user unchecks the lossless checkbox (brining back the quality slider).

What should the quality slider be? 100 or 30?

I feel strongly that it should be 30. The quality slider in the UI is only used for lossy quality control, so it should be returned to the last state the user set it to. They checked "lossless", sure, but I don't think that means the quality slider is set to 100.

So, if we're going to return it to a value of 30, we need to store that state somewhere, and it's stored in the `quality` UI state.

When deriving the `EncodeOptions`, the `quality` UI state is ignored when the intended encode is lossless (see the code in `_inputChange`).

`getDerivedStateFromProps` performs the initial conversion from encode state to UI state. If the incoming encode options are lossless, then we need to figure out what to set the quality slider to if it's shown, hence:

```ts
const quality = lossless ? defaultOptions.quality : options.quality;
```

This code sets the quality slider to `defaultOptions.quality` if the incoming encode options are lossless. This avoids the quality slider being set to 100, which doesn't make sense as a default for a slider related to lossy quality.

Reply:
@jakearchibald  Ah right, if it's lossless then the quality should just be whatever it was before and the encoder should be given `100` without changing the state.

On a related note, I think

```typescript
const quality = lossless ? defaultOptions.quality : options.quality;
```

is redundant anyway because we're using `getDerivedStateFromProps()` only when the AVIF codec is selected, so `options` will be equal to `defaultOptions`. I've removed this line in the latest commit and it still (correctly) preserves the original `quality` before toggling lossless mode.

Reply:
Compare these steps on this deploy vs live:

1. Open demo image
2. On the right-hand side, select AVIF
3. On the right-hand side, select lossless
4. Copy settings to the left side (using the [< >] icon)
5. On the left-hand side, deselect lossless

On squoosh.app this gives you a default quality setting. In this PR it gives you a quality of 100.

I'm pretty sure this is because you removed that line, which would use `defaultOptions.quality` if the initial settings are lossless.